boa_engine/src/vm/code_block.rs

```diff
@@ -252,6 +250,33 @@ impl CodeBlock {
             .rev()
             .find(|(_, handler)| handler.contains(pc))
     }
+
+    pub(crate) fn constant_string_expect(&self, index: usize) -> JsString {
+        if let Constant::String(value) = &self.contants[index] {
+            return value.clone();
+        }
+
+        unreachable!("there should be a string constant at index {index}")
```

Comment:
I'd change these for panics. `unreachable` gives the idea that the function has asserted that `index` points to a `Constant::String` and every other type of constant is impossible to reach, which isn't really true.